indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -168,20 +180,24 @@ def _process(self, action, comment):
 class RHReplaceRevision(RHContributionEditableRevisionBase):
     """Replace the latest revision of an Editable."""
 
+    SERVICE_ALLOWED = True
+
     def _check_revision_access(self):
         # XXX: we still need UI for this or expose it only to the microservice
         return self.editable.editor is None and self.editable.can_perform_submitter_actions(session.user)
 
     @use_kwargs({
-        'comment': fields.String(missing='')
+        'comment': fields.String(missing=''),
+        'state': fields.String(validate=validate.OneOf(InitialRevisionState.__members__))
```

Comment:
```suggestion
        'state': EnumField(InitialRevisionState)
```

Reply:
I had the impression there was a better way of doing this 😄